WordPress/UITestsFoundation/Screens/Me/SupportScreen.swift

```diff
@@ -36,6 +44,21 @@ public class SupportScreen: ScreenObject {
         )
     }
 
+    public func contactSupport() throws -> ContactUsScreen {
+        app.cells["contact-support-button"].tap()
+        addContactInformationIfNeeded()
```

Comment:
In order to contact the support the user must add contact information. When reaching out for the first time, a pop-up is displayed after tapping the `Contact Support` button asking to enter (for logged out users) or confirm (for logged in users). The next time the user tries to contact support the application uses the stored information, so no pop-up is displayed. So the only way to know if addin the contact information is needed is checking if the pop-up is displayed. The  `...IfNeeded` comes to give some flexibility so whoever is writing a test case doesn't need to care about it.

> i think if we know the condition, we have the control to call it only on tests that need it. on tests that don't, we can exclude it and not wait for the condition to happen. wdyt?

Right now we only have one test covering the Contact Support and we will always need to inform the contact information because 1) the current test covers the user logged out case and 2) since we've been removing and installing the app before each test, even if the test covered a logged in case, it would still see the pop-up. However, if we need to write a new tests that visit the Support screen twice, the second one wouldn’t see the pop-up.

I'm not sure if we would gain much? Let me know if I got you wrong... =]

Thank you! 🙇‍♂️

Reply:
thanks for the context and explanation @tiagomar!

while it gives the flexibility of not needing to know the condition, i'm always wary of using something like this as it might bloat the test suite with `waits` in the long run. when we end up with a lot of `...IfNeeded` methods for things we know/can control, for cases where it's not needed, there will be additional waiting time that could have been avoided.

in this case, since there's only 1 test case (for now) and the pop-up seems like it's going to be most likely always needed, what do you think about simplifying it to something like (and dropping `addContactInformationIfNeeded` method):
```
    public func contactSupport(userEmail: String) throws -> ContactUsScreen {
        let emailTextField = app.textFields["Email"]

        app.cells["contact-support-button"].tap()
        // email and username pop up displayed on first visit
        emailTextField.tap()
        emailTextField.typeText(userEmail)
        app.buttons["OK"].tap()

        return try ContactUsScreen()
    }

```

i don't think there are any critical scenarios we would want to cover in UI tests where we would need to visit the support screen twice. i could be wrong though! feel free to keep it as is if there are cases that i'm missing/didn't think about.

Reply:
> i don't think there are any critical scenarios we would want to cover in UI tests where we would need to visit the support screen twice. i could be wrong though! feel free to keep it as is if there are cases that i'm missing/didn't think about.

You're right here. Given that the pop-up is not network triggered, I don't think removing this wait will bring flakiness in, so let's try it out. Done in [`3483025`](https://github.com/wordpress-mobile/WordPress-iOS/pull/20226/commits/34830251e83292b43af5981aff22529226241a3e).